app/controllers/ResponseController.scala

```diff
@@ -76,11 +229,19 @@ class ResponseController @Inject()(
         .fold(
           hasErrors => renderer.render("response.njk", json(hasErrors)).map(BadRequest(_)),
           (value: ResponseModel) => {
-            println(s"************* ArrivalId ${value.arrivalId} ************")
-            println(s"************* Version ${value.version} *************")
-            println(s"************* messageType ${value.messageType} *************")
-            destinationConnector.goodsReleased(goodsReleasedXml, value.arrivalId, value.version)
-            Future.successful(Redirect(routes.ResponseController.onPageLoad()))
+            value.messageType match {
+              case x if (x == "1" || x == "2") =>
+                destinationConnector.sendMessage(goodsReleasedXml, value.arrivalId, value.version, "IE025")
+                Future.successful(Redirect(routes.ResponseController.onPageLoad()))
+              case "3" =>
+                destinationConnector.sendMessage(unloadingPermissionWithSealsXml, value.arrivalId, value.version, "IE043")
+                Future.successful(Redirect(routes.ResponseController.onPageLoad()))
```

Comment:
case 3 and 4 are almost the same. I suspect we're going to have more examples in the future of unloading permission messages, consider creating a generic method that does the sendMessage and returns are response (only difference between the 2 is the xml were sending)

Reply:
I've refactored this so the matcher returns the xml and we have a single call.  is this better?